### PR TITLE
Add quantized matmul work-area size helper API

### LINKED ISSUE #49: [Feature Request] Add Quantized Matmul Work-Area Size Helper API
Proposed PR branch: `qe/issue-10-quantized-matmul-work-area-size`

Category: `Feature Request` (performance + usability)

### Problem
- Quantized matmul can allocate internal scratch when `work_area == NULL`.
- Callers need a sizing helper for deterministic preallocation.

### Justification / why this is needed
- Quantized matmul is typically used in hot paths; allocator overhead here is
  especially costly.
- Sizing helper allows frameworks to allocate once, reuse, and avoid spikes.

### Proposed change
- Add `zdnn_get_quantized_matmul_work_area_size(...)`.
- Add tests.
- (Optional follow-up) Add README guidance for caller preallocation in quantized matmul hot paths.

### Acceptance criteria
- Helper computes required bytes (return `0` for invalid inputs).
- Tests cover edge cases and representative shapes.

### Test plan
- `make test` (expects `tests/testDriver_quantized_matmul_work_area_size_apis.c` to pass).

### References
- Key files: `zdnn/quantized_matmul_work_area.c`, `zdnn/zdnn.h`, `zdnn/zdnn.map`,
  `tests/testDriver_quantized_matmul_work_area_size_apis.c`.

### PR DESCRIPTION
**Pull request by k8ika0s**
Fixes #49

## Summary
- Add quantized matmul work-area size helper API.

## Why
- See #49 for the full problem statement, justification, and acceptance criteria.

## Changes
- Implemented on branch `qe/issue-10-quantized-matmul-work-area-size`.
- Includes code and tests scoped to this issue.

## Validation
- s390x integrated battery pass recorded in `artifacts/final-validation-20260212T173158Z`.
- Targeted regressions for this scope were validated during branch prep.

## Notes
- DCO signoffs are present on branch commits.

### CODE REVIEW

**Comment by MarkVeerasingam** [comment]
H @k8ika0s, thanks for putting this together!

I'm a recent member to the zDNN group, I support the goal, however there is just a few thoughts and concerns I have re: the abstraction and need of this.

As I understand it, this API would expose zDNN's internal scratch-memory requirement so consumers can manage the allocation themselves. My concern is whether we want to make that implementation detail part of the public API contract.

Do we have any evidence that the internal allocation is a significant bottleneck for quantized matmul? If there are benchmarks or workload data showing that quantized matmul is commonly on a hot path and that the associated allocation overhead has a measurable performance impact, I'd be interested in seeing that.

I think having that data would help us determine whether the direction of this!

File: zdnn/quantized_matmul_work_area.c [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+uint64_t zdnn_get_quantized_matmul_work_area_size(const zdnn_ztensor *input_c,
+                                                  bool pre_computed) {
+  // pre_computed mode does not create qc_tilde and therefore does not require
+  // quantized-matmul work area scratch storage.
+  if (pre_computed) {
+    return 0;
+  }
+
+  if (!input_c || !input_c->transformed_desc) {
```
Comment (MarkVeerasingam) [inline zdnn/quantized_matmul_work_area.c:35]: I think an error/status return would be more appropriate here to avoid a silent fail. Since 0 is also a legitimate result when no scratch memory is required (e.g. pre_computed).

File: zdnn/quantized_matmul_work_area.c [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  // aiu_quantized_matmul() requires input_c transformed type to be int8 when
+  // pre_computed is false.
+  if (input_c->transformed_desc->type != ZDNN_BINARY_INT8) {
+    return 0;
+  }
+
+  // The internal temporary tensor (qc_tilde) reuses input_c's transformed
+  // dimensions/layout/format but promotes the element type to DLFLOAT16.
+  zdnn_tensor_desc qc_tilde_desc;
```
Comment (MarkVeerasingam) [inline zdnn/quantized_matmul_work_area.c:47]: Could we avoid duplicating this sizing logic here?

The actual `qc_tilde` allocation path in `aiu_quantized_matmul.c` (L 2343) already constructs the `qc_tilde_desc` and derives `qc_tilde.buffer_size` using `zdnn_getsize_ztensor()`. 
```c
qc_tilde.buffer_size = zdnn_getsize_ztensor(&qc_tilde_desc);
```
This new API appears to reproduce that logic separately.

Could we reuse/expose this existing sizing logic for maintenance purposes

Comment (k8ika0s) [reply zdnn/quantized_matmul_work_area.c:47]: I no longer have any access to validate or test any of this. It was submitted months ago.
